flutter/lib/camera_feature.dart

```diff
@@ -0,0 +1,162 @@
+  @override
+  void initState() {
+    super.initState();
+    //display the current output from the Camera, create a CameraController.
+    _controller =  CameraController (
+      widget.camera,
+      ResolutionPreset.medium,
+    );
+
+    //initialize the controller; returns a Future.
+    _initializeControllerFuture = _controller.initialize();
+    }
+
+    @override
+    void dispose() {
+      // Dispose of the controller when the widget is disposed.
+      _controller.dispose();
+      super.dispose();
+    }
+
+    @override
+    Widget build(BuildContext context) {
+      return Scaffold(
+        appBar: AppBar(title: const Text('Take a picture')),
+        // FutureBuilder displays a loading spinner until the controller has finished initializing.
+        body: FutureBuilder<void>(
+          future: _initializeControllerFuture,
+          builder: (context, snapshot) {
+            if (snapshot.connectionState == ConnectionState.done) {
+              return CameraPreview(_controller);
+            } else {
+              return const Center(child: CircularProgressIndicator());
+            }
+          },
+        ),
+        floatingActionButton: FloatingActionButton(
+        //onPressed callback for taking a picture
+        onPressed: () async {
+          try {
+            await _initializeControllerFuture;
+
+            // retrieves picture
+            final image = await _controller.takePicture();
+            final path = image.path;
+            final bytes = await File(path).readAsBytes();
+            final img.Image imageToAPI = img.decodeImage(bytes);
+
+            //send image to backend and retrieves emotion
+            // make a Uri from the API
+            var uri = Uri.parse('https://moodspot1.pythonanywhere.com/check');
+
+            // here is our main request
+            var request = http.MultipartRequest('POST', uri)
+              ..files.add(
+                  await http.MultipartFile.fromPath(
+                      'selfiePictureRequest',     // the lable by which you must send the file
+                      imageToAPI // the image file, where ever you store that
+                  )
+              );
+
+            var response = await request.send();
+            var topEmotion;
+
+            if(response["error"] == true) {
+              topEmotion = "none";
+            } else {
+              topEmotion = response["dominant_emotion"];
+            }
+
+
+            if (!mounted) return;
+
+            //display on a new screen
+            await Navigator.of(context).push(
+              MaterialPageRoute(
+                builder: (context) => DisplayPictureScreen(
+                  imagePath: image.path,
+                ),
+              ),
+            );
+          } catch (e) {
+            print(e);
+          }
+        },
+        child: const Icon(Icons.camera_alt),
+        ),
+      );
+    }
+  }
+
+//displays picture
+  class DisplayPictureScreen extends StatelessWidget {
```

Comment:
For better readability, each widget should live in its own file. Move this into another file called `display_picture_screen.dart`.